hopr/hopr-lib/src/chain.rs

```diff
@@ -332,7 +332,7 @@ type ActiveTxExecutor = EthereumTransactionExecutor<
 >;
 
 pub fn build_chain_components<Db>(
-    me_onchain: &ChainKeypair,
+    me_onchain: ChainKeypair,
```

Comment:
Why do you copy it here instead of passing a reference ? `ChainKeypair` is not `Copy` for a reason (it contains a secret) and must be cloned, which is inconvenient.

Reply:
That part is not finished yet. Still under construction and hasn't been tested yet.

Reply:
postponed for https://github.com/hoprnet/hoprnet/issues/6049